subsys/bluetooth/enocean.c

```diff
@@ -378,13 +401,15 @@ static void adv_recv(const struct bt_le_scan_recv_info *info,
 	uint8_t len = net_buf_simple_pull_u8(buf);
```

Comment:
Should there not be a general check to ensure that the length field value aligns with the buf->len?

I imagine the following scenario:

1.  You receive an incoming switch data payload (!has_shortened_name) with a malformed length, E.G length_field_val is greater than buf_len.
2. The incorrect length_field_val is copied to len variable
3.  Before `handle_switch_data` is called, the check verifying that the message is between 12 and 16 passes, while in reality, the remaining buf->len is actually 3 for instance.
4. `handle_switch_data` tries to pull more than 3 bytes from the buffer, and the whole thing crashes.

Am I missing something or is this a real issue?

Reply:
No, I think this is a real issue. I would, however, prefer it to be reported as a separate bug, as what I'm trying to do here is to fix the "module thinks sensor data is commissioning" problem specifically, while trying to keep all other behavior the same (even if "the same" == buggy, i.e. I haven't spent time looking at other potential bugs that are already here)

Reply:
(I don't want this PR to turn into a "fix every bug and problem in enocean.c"-PR)

Reply:
You can do it since you did the read-through of this and can probably write down the bug description better than I can